### PR TITLE
Con 2456 implement search event

### PR DESCRIPTION
**Pull request by kernelkiller**

### CODE REVIEW

File: search/action-creators/errorSearchResults.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { ERROR_SEARCH_RESULTS } from '../constants';
+
+/**
+ * Creates the dispatched RECEIVE_SEARCH_RESULTS action object.
```
Comment (richardgorman) [inline search/action-creators/errorSearchResults.js]: Typo: `RECEIVE_SEARCH_RESULTS`

File: search/actions/getSearchResults.js [hunk 2]
```diff
@@ -19,27 +19,41 @@ import { getSearchPhrase } from '../selectors';
...
-  dispatch(
+  const promise = dispatch(
     getProducts({
       params: {
         searchPhrase,
         offset,
         limit,
-        sort,
+      },
+      onBeforeSend: () => {
```
Comment (richardgorman) [inline search/actions/getSearchResults.js]: Maybe it makes sense to rename this to `onBeforeDispatch`?

**Comment by richardgorman** [comment]
Please watch out for missing periods at the end of comments.

**Comment by richardgorman** [comment]
Please resolve merge conflicts

File: search/actions/getSearchResults.js [hunk 3]
```diff
@@ -19,27 +19,41 @@ import { getSearchPhrase } from '../selectors';
...
+        // Dispatch the request action before the related pipeline request is executed
+        dispatch(requestSearchResults(searchPhrase, offset));
       },
     })
-  ).then(() => {
-    dispatch(receiveSearchResults(searchPhrase));
-  });
+  );
+
+  if (promise instanceof Promise) {
```
Comment (richardgorman) [inline search/actions/getSearchResults.js]: Please add a comment here explaining the need for this check.

File: search/actions/getSearchResults.js [hunk 4]
```diff
@@ -19,27 +19,41 @@ import { getSearchPhrase } from '../selectors';
...
+    promise.then((response) => {
+      const { products } = response;
```
Comment (richardgorman) [inline search/actions/getSearchResults.js]: Personally, I am against destructuring if the variable is only used in one place.

File: search/action-creators/errorSearchResults.js (hunk 1 above)
Comment (kernelkiller) [reply search/action-creators/errorSearchResults.js]: done

File: search/actions/getSearchResults.js (hunk 2 above)
Comment (kernelkiller) [reply search/actions/getSearchResults.js]: changed

File: search/actions/getSearchResults.js (hunk 4 above)
Comment (kernelkiller) [reply search/actions/getSearchResults.js]: removed id

File: search/actions/getSearchResults.js (hunk 3 above)
Comment (kernelkiller) [reply search/actions/getSearchResults.js]: added

**Pull request merged**